src/main/java/com/commercetools/sync/types/utils/TypeUpdateFieldDefinitionActionUtils.java

```diff
@@ -0,0 +1,277 @@
+package com.commercetools.sync.types.utils;
+
+import com.commercetools.sync.commons.exceptions.BuildUpdateActionException;
+import com.commercetools.sync.commons.exceptions.DuplicateKeyException;
+import com.commercetools.sync.commons.exceptions.DuplicateNameException;
+import io.sphere.sdk.commands.UpdateAction;
+import io.sphere.sdk.types.FieldDefinition;
+import io.sphere.sdk.types.Type;
+import io.sphere.sdk.types.commands.updateactions.AddFieldDefinition;
+import io.sphere.sdk.types.commands.updateactions.ChangeFieldDefinitionOrder;
+import io.sphere.sdk.types.commands.updateactions.RemoveFieldDefinition;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static com.commercetools.sync.commons.utils.CommonTypeUpdateActionUtils.buildUpdateAction;
+import static com.commercetools.sync.types.utils.FieldDefinitionUpdateActionUtils.buildActions;
+import static java.lang.String.format;
+import static java.util.Collections.singletonList;
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.Collectors.toMap;
+
+public final class TypeUpdateFieldDefinitionActionUtils {
+
+    /**
+     * Compares a list of {@link FieldDefinition}s with a list of {@link FieldDefinition}s.
+     * The method serves as a generic implementation for field definitions syncing and building the required
+     * update actions (AddFieldDefinition, RemoveFieldDefinition, ChangeFieldDefinitionOrder)
+     * and 1-1 update actions on field definitions (e.g. changeFieldDefinitionLabel, etc..) for the required
+     * resource.
+     *
+     * <p>If the list of new {@link FieldDefinition}s is {@code null}, then remove actions are built for
+     * every existing field definition in the {@code oldFieldDefinitions} list.
+     *
+     * @param oldFieldDefinitions       the old list of field definitions.
+     * @param newFieldDefinitions the new list of field definitions.
+     * @return a list of field definitions update actions if the list of field definitions is not identical.
+     *         Otherwise, if the field definitions are identical, an empty list is returned.
+     * @throws BuildUpdateActionException in case there are field definitions drafts with duplicate names.
+     */
+    @Nonnull
+    public static List<UpdateAction<Type>> buildFieldDefinitionsUpdateActions(
+        @Nonnull final List<FieldDefinition> oldFieldDefinitions,
+        @Nullable final List<FieldDefinition> newFieldDefinitions)
+        throws BuildUpdateActionException {
+
+        if (newFieldDefinitions != null) {
+            return buildUpdateActions(
+                oldFieldDefinitions,
```

Comment:
I think having the method call is less lines (less scrolling in the class) and more readable.

Reply:
Please apply this where possible (as long doesn't exceed max width)

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/304/commits/9801dbd87fa39ffda80ead4f33a0fac5f7103070#diff-e071adde6fef713b55f6b5e2f7e683faR55